config/v1/types_cluster_version.go

```diff
@@ -290,6 +305,93 @@ type Release struct {
 // availableUpdates field is accurate and recent.
 const RetrievedUpdates ClusterStatusConditionType = "RetrievedUpdates"
 
+// ConditionalUpdate represents an update which is recommended to some
+// clusters on the version the current cluster is reconciling, but which
+// may not be recommended for the current cluster.
+// +k8s:deepcopy-gen=true
+type ConditionalUpdate struct {
+	// release is the target of the update.
+	// +required
+	Release Release `json:"release"`
+
+	// risks represents the range of issues associated with
+	// updating to the target release. The cluster-version
+	// operator will evaluate all entries, and only recommend the
+	// update if there is at least one entry and all entries
+	// recommend the update.
+	// +optional
+	Risks []ConditionalUpdateRisk `json:"risks,omitempty"`
+
+	// conditions represents the observations of the conditional update's
+	// current status. Known types are:
+	// * Evaluating, for whether the cluster-version operator will attempt to evaluate any risks[].matchingRules.
+	// * Recommended, for whether the update is recommended for the current cluster.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+}
+
+// ConditionalUpdateRisk represents a reason and cluster-state
+// for not recommending a conditional update.
+// +k8s:deepcopy-gen=true
+type ConditionalUpdateRisk struct {
+	// url contains information about this risk.
+	// +optional
```

Comment:
This should be required. The backing graph must require this information before serving the edge as conditional.  We cannot allow conditional updates to have risks that are not described.

Reply:
> We cannot allow conditional updates to have risks that are not described.

Per [here][1] and [here][2], the goal is that we absolutely do declare all of this metadata for every conditional update / risk declared by Red Hat.  But that the API itself allow for these to be missing, because saying things like "I don't think this is a good idea for you, but am not currently at liberty to tell you why" seems like a thing, even if it's a thing we never do as Red Hatters.  So the process would be:

1. [Graph-data][3] grows presubmits to ensure _we_ always set these.
2. [Cincinnati][4] consumes graph-data and generates Cincinnati JSON, allowing for cases where this optional metadata is not set.
3. [The cluster-version operator][5] consumes Cincinnati JSON, allowing for cases where this optional metadata is not set.

So the guard is in step 1, and I don't see a need to make formal requirements for the later stages.

And I [do see a downside][2] in that rejecting metadata-less conditions in step 3 and having the CVO say "hey, your `upstream` has conditions that don't link a URL!  I'm not going to tell you anything about recommended updated until you fix that!" seems strictly worse than that CVO saying "updating to `$VERSION` isn't recommended for your cluster.  I dunno why, because your `upstream` didn't link a URL".

[1]: https://github.com/openshift/enhancements/pull/821#discussion_r708564252
[2]: https://github.com/openshift/enhancements/blob/e539d55691f2710d4fae0a34ac7c4b6b4e87fa10/enhancements/update/targeted-update-edge-blocking.md#metadata-to-include-when-blocking-a-conditional-request
[3]: https://github.com/openshift/cincinnati-graph-data
[4]: https://github.com/openshift/cincinnati
[5]: https://github.com/openshift/cluster-version-operator

Reply:
> because saying things like "I don't think this is a good idea for you, but am not currently at liberty to tell you why" seems like a thing, even if it's a thing we never do as Red Hatters. So the process would be:

This doesn't seem like a reasonable thing for us to do.



> And I [do see a downside](https://github.com/openshift/enhancements/blob/e539d55691f2710d4fae0a34ac7c4b6b4e87fa10/enhancements/update/targeted-update-edge-blocking.md#metadata-to-include-when-blocking-a-conditional-request) in that rejecting metadata-less conditions in step 3 and having the CVO say "hey, your `upstream` has conditions that don't link a URL! I'm not going to tell you anything about recommended updated until you fix that!"

I would rather not have a record of the update at all, than have data that produces unanswerable questions.

Reply:
I still think this is strictly worse than leaving these fields optional, but I'm out of time to argue my case.  So I've pushed 7af2373d -> a93e6749 to pivot to all of your requested `+required`.